mips_api/__init__.py

```diff
@@ -30,42 +171,55 @@ def lambda_handler(event, context):
         Return doc: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html
     """
 
-    # get cache-control max-age from environment variable
-    cache_age = None
-    try:
-        cache_age = os.environ['CacheTTL']
-    except KeyError as exc:
-        return {
-            "statusCode": 500,
-            "body": "Missing environment variable: CacheTTL",
-        }
-
-    # helper function to structure return data and set cache control
+    # helper functions to encapsulate the body, headers, and status code
     def _build_return(code, body):
         return {
             "statusCode": code,
-            "body": json.dumps(body),
-            "headers": {"cache-control": f"max-age={cache_age}"},
+            "body": json.dumps(body, indent=2),
         }
 
-    # get secure parameters
-    global mips_app
     try:
-        if mips_app is None:
-            mips_app = mips.App()
-        mips_app.collect_secrets()
-    except Exception as exc:
-        return _build_return(500, {"error": str(exc)})
+        # collect environment variables
+        mips_org = _get_os_var('MipsOrg')
+        ssm_path = _get_os_var('SsmPath')
+
+        api_routes = {}
+        api_routes['ApiChartOfAccounts'] = _get_os_var('ApiChartOfAccounts')
+        api_routes['ApiValidTags'] = _get_os_var('ApiValidTags')
+
+        _to_omit = _get_os_var('CodesToOmit')
+        omit_codes_list = _parse_omit_codes(_to_omit)
+
+        _to_add = _get_os_var('CodesToAdd')
+        extra_codes_dict = _parse_extra_codes(_to_add)
+
+        # get secure parameters
+        ssm_secrets = collect_secrets(ssm_path)
 
-    # parse the path and get the data
-    if 'path' in event:
-        event_path = event['path']
+        # get chart of accounts from mips
+        mips_chart = collect_chart(mips_org)
 
-        try:
-            mips_data = mips_app.get_mips_data(event_path)
-        except Exception as exc:
-            return _build_return(500, {"error": str(exc)})
+        # parse the path and return appropriate data
+        if 'path' in event:
+            event_path = event['path']
 
-        return _build_return(200, mips_data)
+            if event_path == api_routes['ApiChartOfAccounts']:
+                # return chart of accounts
+                return _build_return(200, mips_chart)
 
-    return _build_return(400, {"error": f"Invalid event: No path found: {event}"})
+            elif event_path == api_routes['ApiValidTags']:
+                try:
+                    valid_tags = list_tags(omit_codes_list, extra_codes_dict)
+                except Exception as exc:
+                    return _build_return(500, {"error": str(exc)})
+
+                # return valid tags
+                return _build_return(200, valid_tags)
+
+            else:
+                return _build_return(404, {"error": "Invalid request path"})
+
+        return _build_return(400, {"error": f"Invalid event: No path found: {event}"})
```

Comment:
not a big deal but i think refactoring this into a function would make testing easier.